### PR TITLE
chore(EventEmitter): add useEventEmitter

### PR DESCRIPTION
**Pull request by tujoworker**
- For internal usage only as of now
- Rewrite `EventEmitter` to TypeScript
- Move it into `/helpers`
- add `useEventEmitter` React Hook + Story Book example

Here is a [CSB](https://codesandbox.io/s/eufemia-event-emitter-example-t2dtut?file=/src/App.tsx)

### CODE REVIEW

File: packages/dnb-eufemia/src/shared/helpers/__tests__/useEventEmitter.test.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+  it('has update function', () => {
+    const { result } = renderHook(() => {
+      return useEventEmitter('unique-id')
+    })
+
+    expect(typeof result.current.update).toBe('function')
+  })
+
+  it('has will set data', () => {
```
Comment (AadneRo) [inline packages/dnb-eufemia/src/shared/helpers/__tests__/useEventEmitter.test.tsx:25]: Can we rename this test description? maybe "It stores the value when updating"

**Review by AadneRo — COMMENTED** [review]
Except from my one comment this looks good to me, I don't understand everything though 😂

File: packages/dnb-eufemia/src/shared/helpers/__tests__/useEventEmitter.test.tsx (hunk 1 above)
Comment (tujoworker) [reply packages/dnb-eufemia/src/shared/helpers/__tests__/useEventEmitter.test.tsx]: 🙌 Yes, very good point – I added some more tests as well.

**Pull request merged**